src/base.js

```diff
@@ -367,18 +361,17 @@ export default class KintoClientBase {
       // from within a batch operation.
       const msg = "This result is generated from within a batch " +
         "operation and should not be consumed.";
-      return Promise.resolve(raw ? { json: msg, headers: { get() {} } } : msg);
+      return raw ? { json: msg, headers: { get() {} } } : msg;
     }
-    const promise = this.fetchServerSettings().then(_ => {
-      return this.http.request(this.remote + request.path, {
-        ...request,
-        body: stringify ? JSON.stringify(request.body) : request.body,
-      });
+    await this.fetchServerSettings();
```

Comment:
`execute` is used by every other method performing an HTTP query to the server, and I didn't want to call that function in each of them (especially as many are decorated with the `@capable` decorator, which needs server capabilities from the settings). This is basically a way to initialize the client with server settings once and for all (the `fetchServerSettings` memoizes the settings the first time it knows about them). 

How would you do it? Also, it sounds like we should comment this part of the code a little better.